Data/ByteString/Lazy.hs

```diff
@@ -1344,6 +1424,12 @@ interact transformer = putStr . transformer =<< getContents
 -- ---------------------------------------------------------------------
 -- Internal utilities
 
+-- Required for rewrite rules for 'compareLength'
+negateOrdering :: Ordering -> Ordering
+negateOrdering LT = GT
+negateOrdering EQ = EQ
+negateOrdering GT = LT
+
```

Comment:
This function looks redundant to me, it is just `compare EQ`.

Reply:
Should I open a miniature pull request to delete `negateOrdering` and replace its call site with `compare EQ`?

Reply:
Yes, much appreciated. Could you please also combine all rules for `compareLength` under a single `RULES` pragma?